modules/standard/BigInteger.chpl

```diff
@@ -2767,88 +2825,95 @@ module BigInteger {
       return isPrime;
   }
 
-  proc bigint.nextprime(const ref a: bigint) {
+  proc nextprime(ref result: bigint, const ref a: bigint) {
     if _local {
-      mpz_nextprime(this.mpz, a.mpz);
-
-    } else if this.localeId == chpl_nodeID &&
-              a.localeId    == chpl_nodeID {
-      mpz_nextprime(this.mpz, a.mpz);
-
+      mpz_nextprime(result.mpz, a.mpz);
+    } else if result.localeId == chpl_nodeID &&
+              a.localeId == chpl_nodeID {
+      mpz_nextprime(result.mpz, a.mpz);
     } else {
-      const thisLoc = chpl_buildLocaleID(this.localeId, c_sublocid_any);
-
-      on __primitive("chpl_on_locale_num", thisLoc) {
-        var a_ = a;
-
-        mpz_nextprime(this.mpz, a_.mpz);
+      const resultLoc = chpl_buildLocaleID(result.localeId, c_sublocid_any);
+      on __primitive("chpl_on_locale_num", resultLoc) {
+        const a_ = a.localize();
+        mpz_nextprime(result.mpz, a_.mpz);
       }
     }
   }
 
+  @deprecated(notes="bigint.nextprime method is deprecated - please use the standalone function :proc:`~BigInteger.nextprime`")
+  proc bigint.nextprime(const ref a: bigint) {
+    BigInteger.nextprime(this, a);
+  }
 
-
-  // gcd
-  proc bigint.gcd(const ref a: bigint, const ref b: bigint) {
+  proc gcd(ref result: bigint, const ref a: bigint, const ref b: bigint) {
     if _local {
-      mpz_gcd(this.mpz, a.mpz, b.mpz);
-
-    } else if this.localeId == chpl_nodeID &&
+      mpz_gcd(result.mpz, a.mpz, b.mpz);
+    } else if result.localeId == chpl_nodeID &&
               a.localeId    == chpl_nodeID &&
               b.localeId    == chpl_nodeID {
-      mpz_gcd(this.mpz, a.mpz, b.mpz);
-
+      mpz_gcd(result.mpz, a.mpz, b.mpz);
```

Comment:
Looking at this three-pronged check and now that we have localize, I wonder if this could be something like

```chpl
if result.localeId == chpl_nodeID {
  a_ = a.localize();
  b_ = b.localize();
  mpz_gcd(result.mpz, a_.mpz, b_.mpz);
}
```

Which would result in less taking of the `else` path, which I think would be preferable here.